applications/FluidDynamicsApplication/custom_utilities/mass_conservation_utility.cpp

```diff
@@ -246,75 +317,97 @@ void MassConservationCheckProcess::ComputeVolumesAndInterface( double& positiveV
             }
         }
     }
-    // assigning the values to the arguments of type reference
-    positiveVolume = pos_vol;
-    negativeVolume = neg_vol;
-    interfaceArea = int_area;
+    mrModelPart.GetCommunicator().SynchronizeCurrentDataToMin(DISTANCE);
+    rPositiveVolume = mrModelPart.GetCommunicator().GetDataCommunicator().SumAll(pos_vol);
+    rNegativeVolume = mrModelPart.GetCommunicator().GetDataCommunicator().SumAll(neg_vol);
+    rInterfaceArea = mrModelPart.GetCommunicator().GetDataCommunicator().SumAll(int_area);
 }
 
 
-double MassConservationCheckProcess::ComputeInterfaceArea(){
 
-    double int_area = 0.0;
+double MassConservationUtility::OrthogonalFlowIntoAir( const double Factor )
+{
+    double outflow = 0.0;
 
-    #pragma omp parallel for reduction(+: int_area)
+    KRATOS_ERROR_IF((std::abs( Factor ) - 1.0) > 1.0e-12) << "MassConservationPocess: Given value of argument 'factor' is not plausible. Consider '1' or '-1'." << std::endl;
```

Comment:
I'd rather pass an `enum` that takes +1 and -1 as values. In that way you ensure the values will be always +1 and -1.